components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pap/store/PAPPolicyStoreReader.java

```diff
@@ -188,125 +126,42 @@ public boolean isExistPolicy(String policyId) {
 
 
     /**
-     * Reads Light Weight PolicyDTO for given policy id
+     * Reads light weight PolicyDTO for given policy id
      *
      * @param policyId policy id
-     * @return PolicyDTO but don not contains XACML policy and attribute meta data
+     * @return PolicyDTO but does not contain XACML policy and attribute meta data
      * @throws EntitlementException throws, if fails
      */
     public PolicyDTO readLightPolicyDTO(String policyId) throws EntitlementException {
 
-        Resource resource = null;
-        PolicyDTO dto = null;
-        resource = store.getPolicy(policyId, PDPConstants.ENTITLEMENT_POLICY_PAP);
-        if (resource == null) {
+        PolicyDTO dto = store.getPAPPolicy(policyId);
+        if (dto == null) {
             return null;
         }
-        dto = new PolicyDTO();
-        dto.setPolicyId(policyId);
-        String version = resource.getProperty(PDPConstants.POLICY_VERSION);
-        if (version != null) {
-            dto.setVersion(version);
-        }
-        String lastModifiedTime = resource.getProperty(PDPConstants.LAST_MODIFIED_TIME);
-        if (lastModifiedTime != null) {
-            dto.setLastModifiedTime(lastModifiedTime);
-        }
-        String lastModifiedUser = resource.getProperty(PDPConstants.LAST_MODIFIED_USER);
-        if (lastModifiedUser != null) {
-            dto.setLastModifiedUser(lastModifiedUser);
-        }
-        dto.setActive(Boolean.parseBoolean(resource.getProperty(PDPConstants.ACTIVE_POLICY)));
-        String policyOrder = resource.getProperty(PDPConstants.POLICY_ORDER);
-        if (policyOrder != null) {
-            dto.setPolicyOrder(Integer.parseInt(policyOrder));
-        } else {
-            dto.setPolicyOrder(0);
-        }
-        dto.setPolicyType(resource.getProperty(PDPConstants.POLICY_TYPE));
-
-        String policyReferences = resource.getProperty(PDPConstants.POLICY_REFERENCE);
-        if (policyReferences != null && policyReferences.trim().length() > 0) {
-            dto.setPolicyIdReferences(policyReferences.split(PDPConstants.ATTRIBUTE_SEPARATOR));
-        }
-
-        String policySetReferences = resource.getProperty(PDPConstants.POLICY_SET_REFERENCE);
-        if (policySetReferences != null && policySetReferences.trim().length() > 0) {
-            dto.setPolicySetIdReferences(policySetReferences.split(PDPConstants.ATTRIBUTE_SEPARATOR));
-        }
-
-        dto.setPolicyEditor(resource.getProperty(PDPConstants.POLICY_EDITOR_TYPE));
-
+        dto.setPolicy(null);
```

Comment:
Have to validate, whether this approach (pulling the policy from the database and setting the policy as null in the runtime) will actually serve the purpose of building "Light weight" policy DTO.

Reply:
Previously also we anyway load the full resource from registry and only set a subset of the attributes in the DTO. Since we are loading the full object, with regards to registry impl there might not be a considerable performance hit. But we could optimize this by only querying the required data in the db impl to be introduced later, but this will be a complexity overhead.
So, IMO it won't have much of a performance hit, but should analyze if having this light-weight policy DTO concept makes sense anymore.